plugin.rb

```diff
@@ -344,82 +345,72 @@ def self.skip_db?
     topic&.custom_fields&.[](::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD).present?
   end
 
-  #TODO Remove when plugin is 1.0
-  if Search.respond_to? :advanced_filter
-    Search.advanced_filter(/status:solved/) do |posts|
-      posts.where(
-        "topics.id IN (
-        SELECT tc.topic_id
-        FROM topic_custom_fields tc
-        WHERE tc.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' AND
-                        tc.value IS NOT NULL
-        )",
+  solved_callback = ->(scope) do
```

Comment:
We have several ways to filter by solved/unsolved status (in search, in filter, in topic lists, etc...).

I've extracted both solved and unsolved "queries" so they can be used everywhere.